doc/rtd/topics/datasources/vmware.rst

```diff
@@ -7,13 +7,88 @@ This datasource is for use with systems running on a VMware platform such as
 vSphere and currently supports the following data transports:
 
 
+* `Guest OS Customization <https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-58E346FF-83AE-42B8-BE58-253641D257BC.html>`_
 * `GuestInfo <https://github.com/vmware/govmomi/blob/master/govc/USAGE.md>`_ keys
 
 Configuration
 -------------
 
 The configuration method is dependent upon the transport:
 
+Guest OS Customization
+^^^^^^^^^^^^^^^^^^^^^^
+
+The following configuration can be set for this datasource in cloud-init
+configuration (in `/etc/cloud/cloud.cfg` or `/etc/cloud/cloud.cfg.d/`).
+
+System configuration:
+
+``disable_vmware_customization``:
+true(disable) or false(enable) the vmware customization. (default: true)
+
+Datasource configuration:
+
+``allow_raw_data``:
+true(enable) or false(disable) the vmware customization using cloud-init
+metadata and userdata directly. Since vSphere 7.0 Update 3 version, user
+can create a Linux customizatino specification with barely cloud-init
+metadata and userdata and apply this kind specification to a virtual
+machine, this datasource will parse the metadata and userdata and
+configure virtual machine with them. See `Guest Customization Using
+cloud-init <https://developer.vmware.com/docs/14558/vsphere-web-services-sdk-programming-guide--7-0-update-3-/GUID-75E27FA9-2E40-4CBF-BF3D-22DCFC8F11F7.html>`_ (default: true)
+ 
+``vmware_cust_file_max_wait``:
+The maximum amount of clock time in seconds that should be spent waiting
+for vmware customization files. (default: 15)
+
+Configuration examples:
+
+1. Create a file /etc/cloud/cloud.cfg.d/99-vmware-guest-customization.cfg with
+   below content will enable vmware customization, and set the maximum time of
+   waiting for vmware customization file to 10 seconds:
+
+.. code-block:: yaml
+
+   disable_vmware_customization: false
+   datasource:
+     VMware:
+       vmware_cust_file_max_wait: 10
+
+2. Create a file /etc/cloud/cloud.cfg.d/99-vmware-guest-customization.cfg with
+   below content will enable vmware customization, but only try to apply a
+   traditional Linux guest customization configuration:
+
+.. code-block:: yaml
+
+   disable_vmware_customization: false
+   datasource:
+     VMware:
+       allow_raw_data: false
```

Comment:
Can we keep vmware_cus_file_max_wait after this line so that user could know this flag works for the 2 types of customization.

Reply:
Yes, made this change in the last commit.